src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -244,9 +244,8 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                     if (inputType.IsNullableType() && (convertedExpression.ConstantValue == null || !convertedExpression.ConstantValue.IsNull))
                     {
                         // Null is a special case here because we want to compare null to the Nullable<T> itself, not to the underlying type.
-                        var discardedDiagnostics = DiagnosticBag.GetInstance(); // We are not interested in the diagnostic that get created here
-                        convertedExpression = CreateConversion(operand, inputType.GetNullableUnderlyingType(), discardedDiagnostics);
-                        discardedDiagnostics.Free();
+                        // We are not interested in the diagnostic that get created here
+                        convertedExpression = CreateConversion(operand, inputType.GetNullableUnderlyingType(), BindingDiagnosticBag.Discarded);
```

Comment:
This line of the comment could probably be removed. It's obvious we're not interested in the diagnostic, we're using `BindingDiagnosticBad.Discarded`

Reply:
> This line of the comment could probably be removed. It's obvious we're not interested in the diagnostic, we're using BindingDiagnosticBad.Discarded

Comment do not hurt and I tried  to keep them.

---
In reply to: [361552733](https://github.com/dotnet/roslyn/pull/40560#discussion_r361552733) [](ancestors = 361552733)